packages/truffle-events/Subscriber/index.js

```diff
@@ -0,0 +1,57 @@
+const helpers = require("./helpers");
+const { createLookupTable, sortHandlers, validateOptions } = helpers;
+
+class Subscriber {
+  constructor({ emitter, options }) {
+    validateOptions(options);
+    const { initialization, handlers } = options;
+
+    this.emitter = emitter;
+
+    if (initialization) initialization.bind(this)();
+
+    const { globbedHandlers, nonGlobbedHandlers } = sortHandlers(handlers);
+
+    if (nonGlobbedHandlers) this.setUpListeners(nonGlobbedHandlers);
+
+    if (globbedHandlers) {
+      this.globbedHandlers = globbedHandlers;
+      this.setUpGlobbedListeners(globbedHandlers);
+    }
+  }
+
+  handleEvent(eventName, data) {
+    for (let handlerName in this.globbedHandlerLookupTable) {
+      if (this.regexMatchesEntireName(eventName, handlerName)) {
+        this.globbedHandlers[handlerName].forEach(handler => {
+          handler.bind(this, data)();
```

Comment:
Why do we need to pass the event name into this? I don't think the Subscriber needs to know about that...or at least the eventName is passed into the above function as an argument and that gets matched against the regexes in the lookup table.

However, looking at the code above, it does seem like it should actually be `handler.bind(this)(data)`, right?

Reply:
My thinking is that a Subscriber bound via a wildcard event (.e.g, "namespace:**") would have no way of knowing what event was being triggered. it could see imagine a case where the subscriber would want to react to the event name, or maybe just log it.

I prefer `handler.bind(this)(data)` over `handler.bind(this, data)()`, but ultimately it'll work either way and don't think readability would be improved by changing it.